backend-rust/src/graphql_api.rs

```diff
@@ -3466,26 +3547,171 @@ impl Account {
 
     async fn account_statement(
         &self,
-        #[graphql(desc = "Returns the first _n_ elements from the list.")] first: i32,
+        ctx: &Context<'_>,
+        #[graphql(desc = "Returns the first _n_ elements from the list.")] first: Option<u64>,
         #[graphql(desc = "Returns the elements in the list that come after the specified cursor.")]
-        after: String,
-        #[graphql(desc = "Returns the last _n_ elements from the list.")] last: i32,
+        after: Option<String>,
+        #[graphql(desc = "Returns the last _n_ elements from the list.")] last: Option<u64>,
         #[graphql(desc = "Returns the elements in the list that come before the specified cursor.")]
-        before: String,
+        before: Option<String>,
     ) -> ApiResult<connection::Connection<String, AccountStatementEntry>> {
-        todo_api!()
+        let config = get_config(ctx)?;
+        let pool = get_pool(ctx)?;
+        let query = ConnectionQuery::<i64>::new(
+            first,
+            after,
+            last,
+            before,
+            config.account_statements_connection_limit,
+        )?;
+
+        let mut account_statements = sqlx::query_as!(
+            AccountStatementEntry,
+            r#"
+            SELECT
```

Comment:
I think we are missing the subquery for handling the `last` query parameter, currently it seems that it will behave identically to providing `first` as we are not ordering a subquery according to `query.desc` (which is true when `last` is provided, the bad naming of `query.desc` is on me)